chainer/variable.py

```diff
@@ -1562,19 +1587,11 @@ def add_cand(cand):
 
     grads = _backprop_utils.GradTable(load_if_new=True)
 
-    root_nodes = set()
     leaf_nodes = set()
 
-    for y_var in outputs:
-        # TODO(sonots): Implement for ChainerX
-        if y_var.xp is chainerx:
-            raise NotImplementedError()
-
-        y = y_var.node
-        root_nodes.add(y)
-        grads[y] = y_var.grad_var
+    for y, gy in outputs:
+        grads.get_as_list(y).append(gy)
```

Comment:
How about creating a new method in `GradTable` for this?
This line looks unusual and perhaps surprising that `append` affects `grads`.